src/ConditionsResolver/Resovler/ConditionalGroup.cs

```diff
@@ -0,0 +1,326 @@
+            }
+
+            if (LeftConditional is not null && RightConditional is not null)
+            {
+                return LeftConditional.Evaluate() || RightConditional.Evaluate();
+            }
+            if (LeftGroup is not null && RightGroup is not null)
+            {
+                return LeftGroup.Evaluate() || RightGroup.Evaluate();
+            }
+            if (LeftGroup is not null && RightConditional is not null)
+            {
+                return LeftGroup.Evaluate() || RightConditional.Evaluate();
+            }
+            if (LeftConditional is not null && RightGroup is not null)
+            {
+                return LeftConditional.Evaluate() || RightGroup.Evaluate();
+            }
+
+            throw new InvalidOperationException("Evaluation Error in EvaluateOrs");
+        }
+
+        private bool EvaluteOrsLogicalGroups()
+        {
+            if (LeftConditional is not null && RightGroup is not null)
+            {
+                return RightGroup.Evaluate() || LeftConditional.Evaluate();
+            }
+            if (LeftGroup is not null && RightGroup is not null)
+            {
+                return RightGroup.Evaluate() || LeftGroup.Evaluate();
+            }
+            return false;
+        }
+
+        private bool EvaluateAnds()
+        {
+            if (RightGroup is not null && RightGroup.GroupedLogical > GroupedLogical)
+            {
+                return EvaluteAndsLogicalGroups();
+            }
+
+            if (LeftConditional is not null && RightConditional is not null)
+            {
+                return LeftConditional.Evaluate() && RightConditional.Evaluate();
+            }
+            if (LeftGroup is not null && RightGroup is not null)
+            {
+                return LeftGroup.Evaluate() && RightGroup.Evaluate();
+            }
+            if (LeftGroup is not null && RightConditional is not null)
+            {
+                return LeftGroup.Evaluate() && RightConditional.Evaluate();
+            }
+            if (LeftConditional is not null && RightGroup is not null)
+            {
+                return LeftConditional.Evaluate() && RightGroup.Evaluate();
+            }
+
+            throw new InvalidOperationException("Evaluation Error in EvaluateAnds");
+        }
+
+        private bool EvaluteAndsLogicalGroups()
+        {
+            if (LeftConditional is not null && RightGroup is not null)
+            {
+                return RightGroup.Evaluate() && LeftConditional.Evaluate();
+            }
+            if (LeftGroup is not null && RightGroup is not null)
+            {
+                return RightGroup.Evaluate() && LeftGroup.Evaluate();
+            }
+            return false;
+        }
+
+        public static ConditionalGroup Create(string input, int groupedLogicalParent = 0)
+        {
+            var conditionalGroup = new ConditionalGroup();
+            if (string.IsNullOrEmpty(input))
```

Comment:
May save one initialization call by moving the previous line down when input is null, empty or whitespace.